sound/soc/sof/topology.c

```diff
@@ -1423,29 +1423,29 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		goto widget_free;
 	}
 
-	if (swidget->num_sink_pins > SOF_WIDGET_MAX_NUM_PINS ||
-	    swidget->num_source_pins > SOF_WIDGET_MAX_NUM_PINS) {
-		dev_err(scomp->dev, "invalid pins for %s: [sink: %d, src: %d]\n",
-			swidget->widget->name, swidget->num_sink_pins, swidget->num_source_pins);
+	if (swidget->num_input_pins > SOF_WIDGET_MAX_NUM_PINS ||
+	    swidget->num_output_pins > SOF_WIDGET_MAX_NUM_PINS) {
```

Comment:
should we check that the sum is < SOF_WIDGET_MAX_NUM_PINS

Reply:
the definition of MAX_NUM_PINS applies individually for input and output pins. So I dont think a check for sum is required.

Reply:
didn't we have an array where we first have the input pins and then the output pins?

Reply:
I am pretty sure we discussed this with @aiChaoSONG but I don't know if this applies in this context.
This could be addressed as a separate patch if needed.

Reply:
yes, this is documented here:
https://github.com/thesofproject/linux/pull/4163/commits/cf6e9a5ff0ee4948cd51780459a23382a7c07e21#diff-3aa89cf77fc07290f36500245dad57c983ae8628a8fb4b43804391a01ca4f520R444

But I am not sure what the binding array size might be.